pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -1050,6 +1050,30 @@ func (c *K8sOrchestrator) IsFSSEnabled(ctx context.Context, featureName string)
 			"Setting the feature state to false", featureName, c.internalFSS.configMapName)
 		return false
 	} else if c.clusterFlavor == cnstypes.CnsClusterFlavorWorkload {
+		// Check if it is WCP defined feature state.
+		if _, exists := common.WCPFeatureStates[featureName]; exists {
+			log.Infof("Feature %q is a WCP defined feature state. Reading the %q configmap in %q namespace.",
+				featureName, common.WCPFSSConfigMapName, common.KubeSystemNamespace)
+			// Check the `wcp-cluster-capabilities` configmap in supervisor for the FSS value.
+			wcpFssMap, err := c.k8sClient.CoreV1().ConfigMaps(common.KubeSystemNamespace).Get(ctx,
```

Comment:
Can we cache - wcpFssMap?
So if we are calling `IsFSSEnabled` multiple times we don't end up checking Kubernetes configmap for every call to check fss is enabled or disabled?

Reply:
Cached the `wcpCapabilityFssMap` first time `IsFSSEnabled` is called. I have already reduced the number of calls to `IsFSSEnabled` in both controller and syncer and the caching will be an added benefit.